staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go

```diff
@@ -184,3 +184,65 @@ func ValidateManagedFields(fieldsList []metav1.ManagedFieldsEntry, fldPath *fiel
 	}
 	return allErrs
 }
+
+func ValidateConditions(conditions []metav1.Condition, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	conditionTypeToFirstIndex := map[string]int{}
+	for i, condition := range conditions {
+		if firstIndex, ok := conditionTypeToFirstIndex[condition.Type]; ok {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("type"), condition.Type, fmt.Sprintf("type was used by %v", fldPath.Index(firstIndex).Child("type"))))
+		} else {
+			conditionTypeToFirstIndex[condition.Type] = i
+		}
+
+		allErrs = append(allErrs, ValidateCondition(condition, fldPath.Index(i))...)
+	}
+
+	return allErrs
+}
+
+// validConditionStatuses is used internally to check validity and provide a good message
+var validConditionStatuses = sets.NewString(string(metav1.ConditionTrue), string(metav1.ConditionFalse), string(metav1.ConditionUnknown))
+
+const (
+	maxReasonLen  = 16 * 1024
```

Comment:
this seems excessive for a "programmatic identifier"

the longest reasons I could find in-tree were:
```go
GroupVersionParsingFailed
OverlappingBuiltInResource
InstanceDeletionCompleted
UnknownContainerStatuses
```

something like 64 or 128 characters seems like plenty to me

Reply:
> something like 64 or 128 characters seems like plenty to me

I've seen some longer ones in the hundreds.  Concating during unions as a for instance.  How about 4k?  I want a max value, but I don't want to be constraining to what I've seen already.

Reply:
> I've seen some longer ones in the hundreds. Concating during unions as a for instance. How about 4k? I want a max value, but I don't want to be constraining to what I've seen already.

reduced to 1k